ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -7516,15 +7668,13 @@ OGRLayer *GDALGeoPackageDataset::ExecuteSQL(const char *pszSQLCommand,
         {
             const char *pszSrcTableName = papszTokens[2];
             const char *pszDstTableName = papszTokens[5];
-            OGRGeoPackageTableLayer *poSrcLayer =
-                dynamic_cast<OGRGeoPackageTableLayer *>(
-                    GetLayerByName(SQLUnescape(pszSrcTableName)));
-            if (poSrcLayer)
+            if (!RenameVectorOrRasterLayer(pszSrcTableName, pszDstTableName))
```

Comment:
There might be non-typical use cases where a table would not be recoginized as a vector or raster layers. In those cases, we would want to still forward the ALTER TABLE ... to be executed.

Example:
```
ogr2ogr poly.gpkg autotest/ogr/data/poly.shp
ogrinfo poly.gpkg -sql "create table my_hidden_table(id integer primary key)"
ogrinfo poly.gpkg  -oo LIST_ALL_TABLES=NO -sql "alter table my_hidden_table rename to my_hidden_table_renamed"
```
We need a new test case for that

Reply:
done